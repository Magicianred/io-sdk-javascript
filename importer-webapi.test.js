const fs = require('fs')
const importerWebApi = require("./importer-webapi")

test("importer WebApi", async () => {
    let endpoint = "https://raw.githubusercontent.com/Magicianred/io-sdk-javascript/webapi/data/messages.json";
    const imp = await importerWebApi(endpoint)
    expect(imp != null).toBe(true)
    expect(Array.isArray(imp)).toBe(true)
    expect(imp).toMatchSnapshot()
})

test("importer WebApi NotFound", async () => {
    let endpoint = "https://raw.githubusercontent.com/Magicianred/io-sdk-javascript/webapi/data/notfound.json";
    const imp = await importerWebApi(endpoint)
    expect(imp != null).toBe(true)
    expect(Array.isArray(imp)).toBe(false)
    expect(imp).toMatchSnapshot()
})