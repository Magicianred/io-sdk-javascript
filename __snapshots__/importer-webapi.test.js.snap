// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`importer WebApi 1`] = `
Array [
  Object {
    "content": Object {
      "due_date": "2012-04-23T18:25:43.511Z",
      "markdown": "This is a message",
      "subject": "Hello",
    },
    "fiscal_code": "1234567890123456",
    "time_to_live": 3600,
  },
  Object {
    "due_date": "2002-02-02T00:00:00.000Z",
    "fiscal_code": "ISPXNB32R82Y766D",
    "markdown": "# Welcome, Luca Rossi\\\\nThis is a test message.\\\\n.This message was sent using the IO-SDK.\\\\nYour fiscal code is ISPXNB32R82Y766D\\\\nWe hope you are enjoing using IO.\\\\n",
    "subject": "Hi Luca",
    "time_to_live": 3600,
  },
  Object {
    "due_date": "2001-01-01T00:00:00.000Z",
    "fiscal_code": "ISPXNB32R82Y766F",
    "markdown": "# Welcome, Giovanni Rossi\\\\nThis is a test message.\\\\n.This message was sent using the IO-SDK.\\\\nYour fiscal code is ISPXNB32R82Y766F\\\\nWe hope you are enjoing using IO.\\\\n",
    "subject": "Hi Giovanni",
    "time_to_live": 3600,
  },
]
`;

exports[`importer WebApi NotFound 1`] = `"404: Not Found"`;
